From aad3b9659ba36e372ddf213a5fcc10c30960bf07 Mon Sep 17 00:00:00 2001
Message-Id: <aad3b9659ba36e372ddf213a5fcc10c30960bf07.1592119902.git.liu.denton@gmail.com>
In-Reply-To: <cover.1592119902.git.liu.denton@gmail.com>
References: <7e803a2ba9458ce35c657e67323edfe4409205ec.1592055716.git.liu.denton@gmail.com>
	<cover.1592119902.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Sat, 13 Jun 2020 21:53:57 -0400
Subject: [PATCH v2 1/3] remote-curl: use strlen() instead of magic numbers
To: Git Mailing List <git@vger.kernel.org>
Cc: Jeff King <peff@peff.net>

When we are memcpy()ing the length header, we use the magic literal `4`,
representing the length of "0000" and "0001", the packet line length
headers. Use `strlen("000x")` so that we do not have to use the magic
literal.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 remote-curl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/remote-curl.c b/remote-curl.c
index 75532a8bae..5536372b34 100644
--- a/remote-curl.c
+++ b/remote-curl.c
@@ -596,10 +596,10 @@ static int rpc_read_from_out(struct rpc_state *rpc, int options,
 			set_packet_header(buf - 4, *appended);
 			break;
 		case PACKET_READ_DELIM:
-			memcpy(buf - 4, "0001", 4);
+			memcpy(buf - strlen("0001"), "0001", strlen("0001"));
 			break;
 		case PACKET_READ_FLUSH:
-			memcpy(buf - 4, "0000", 4);
+			memcpy(buf - strlen("0000"), "0000", strlen("0000"));
 			break;
 		case PACKET_READ_RESPONSE_END:
 			die(_("remote server sent stateless separator"));
-- 
2.27.0.132.g321788e831

